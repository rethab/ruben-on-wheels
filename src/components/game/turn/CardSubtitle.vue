<template>
  <v-card-subtitle>
    <span :class="lost ? 'text-red' : ''">
      Points: {{ player.points }}, Motivation: {{ player.motivation }}, City:
      {{ player.currentCity }}
    </span>
  </v-card-subtitle>
</template>

<script setup lang="ts">
import type { Player } from "@/services/types";

interface Props {
  player: Player;
  lost?: boolean;
}
withDefaults(defineProps<Props>(), {
  lost: false,
});
</script>
