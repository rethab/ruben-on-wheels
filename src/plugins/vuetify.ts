import "vuetify/styles";
import { createVuetify } from "vuetify";
import { aliases, mdi } from "vuetify/iconsets/mdi";
import * as components from "vuetify/components";
import * as directives from "vuetify/directives";
import "@mdi/font/css/materialdesignicons.css";

export default createVuetify({
  icons: { defaultSet: "mdi", aliases, sets: { mdi } },
  components,
  directives,
});

// Export for test.
export { components, directives };
