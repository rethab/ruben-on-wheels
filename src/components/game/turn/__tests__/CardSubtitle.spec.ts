import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import CardSubtitle from "../CardSubtitle.vue";

import vuetify from "@/plugins/vuetify";

function mountComponent(lost?: boolean) {
  return mount(CardSubtitle as any, {
    props: {
      player: {
        name: "Alice",
        currentCity: "Zurich",
        motivation: 13,
        points: 123,
      },
      lost,
    },
    global: {
      plugins: [vuetify],
    },
  });
}

describe("CardSubtitle", () => {
  it("renders full component if not lost", () => {
    const wrapper = mountComponent();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders full component if lost", () => {
    const wrapper = mountComponent(true);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
