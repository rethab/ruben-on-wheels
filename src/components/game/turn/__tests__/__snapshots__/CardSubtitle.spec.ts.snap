// Vitest Snapshot v1

exports[`CardSubtitle > renders full component if lost 1`] = `"<div class=\\"v-card-subtitle\\"><span class=\\"text-red\\"> Points: 123, Motivation: 13, City: Zurich</span></div>"`;

exports[`CardSubtitle > renders full component if not lost 1`] = `"<div class=\\"v-card-subtitle\\"><span class=\\"\\"> Points: 123, Motivation: 13, City: Zurich</span></div>"`;
